### PR TITLE
Update the documentation

### PR DESCRIPTION
**Pull request by kfox1111**

### CODE REVIEW

File: charts/spire/README.md [hunk 1]
```diff
@@ -7,33 +7,68 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
...
-
 ## Upgrade notes
 
 We only support upgrading one major version at a time. Version skipping isn't supported.
 
+### 0.17.X
+
+
+The SPIFFE OIDC Discovery Provider now has many new TLS options and out of the box defaults to using SPIRE for its certificate. Also, the `spiffe-oidc-discovery-provider.insecureScheme.enabled` flag was removed. If you previously set that option, remove the setting from your values.yaml and see if the new default is suitable for your deployment. If it isn't, please consider one of the other options under `spiffe-oidc-discovery-provider.tls`. If all other options are still unsuitable, you can still enable the previous mode by disabling all tls options. (`spiffe-oidc-discovery-provider.spire.enabled=false`)
+
```
Comment (kfox1111) [inline charts/spire/README.md:71]:
```suggestion

The SPIFFE OIDC Discovery Provider is now enabled by default. If you previously chose to have it off, you can disable it explicitly with `spiffe-oidc-discovery-provider.enabled=false`.
```

File: charts/spire/README.md [hunk 2]
```diff
@@ -7,33 +7,69 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
...
 
-For production installs, please see [the production example](https://github.com/spiffe/helm-charts-hardened/tree/spire-0.16.0/examples/production).
-
 ## Upgrade notes
 
 We only support upgrading one major version at a time. Version skipping isn't supported.
 
+### 0.17.X
+
+
```
Comment (kfox1111) [inline charts/spire/README.md:69]:
```suggestion

```

File: charts/spire/README.md [hunk 3]
```diff
@@ -7,33 +7,67 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
...
+helm upgrade --install -n spire-mgmt spire spire --repo https://spiffe.github.io/helm-charts-hardened/ -f your-values.yaml
 ```
 
-For production installs, please see [the production example](https://github.com/spiffe/helm-charts-hardened/tree/spire-0.16.0/examples/production).
-
 ## Upgrade notes
 
 We only support upgrading one major version at a time. Version skipping isn't supported.
 
+### 0.17.X
```
Comment (kfox1111) [inline charts/spire/README.md:67]:
```suggestion
### 0.17.X

```

File: charts/spire/README.md [hunk 4]
```diff
@@ -7,33 +7,68 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
...
+
+The SPIFFE OIDC Discovery Provider now has many new TLS options and out of the box defaults to using SPIRE for its certificate. Also, the `spiffe-oidc-discovery-provider.insecureScheme.enabled` flag was removed. If you previously set that option, remove the setting from your values.yaml and see if the new default is suitable for your deployment. If it isn't, please consider one of the other options under `spiffe-oidc-discovery-provider.tls`. If all other options are still unsuitable, you can still enable the previous mode by disabling all tls options. (`spiffe-oidc-discovery-provider.spire.enabled=false`)
+
+The SPIFFE OIDC Discovery Provider is now enabled by default. If you previously chose to have it off, you can disable it explicitly with `spiffe-oidc-discovery-provider.enabled=false`.
 ### 0.16.X
```
Comment (kfox1111) [inline charts/spire/README.md:72]:
```suggestion

### 0.16.X
```

**Comment by mrsabath** [comment]
I tested this version on IBM Cloud with Openshift deployment and it works fine.

**Review by mrsabath — APPROVED** [review]
LGTM after testing with Openshift

File: charts/spire/README.md [hunk 5]
```diff
@@ -7,33 +7,76 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
 
 **Homepage:** <https://github.com/spiffe/helm-charts-hardened/tree/main/charts/spire>
 
-## Install notes
+## Install Instructions
 
+### Non Production
 To do a quick non production install suitable for quick testing in something like minikube:
```
Comment (faisal-memon) [inline charts/spire/README.md:13]:
```suggestion
To do a quick install suitable for testing in something like minikube:
```

File: charts/spire/README.md [hunk 6]
```diff
@@ -7,33 +7,76 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
...
+helm upgrade --install -n spire-server spire-crds spire-crds --repo https://spiffe.github.io/helm-charts-hardened/ --create-namespace
+helm upgrade --install -n spire-server spire spire --repo https://spiffe.github.io/helm-charts-hardened/
 ```
 
-To customize, start with a base values file and edit as needed:
+### Production
+
+Preparing a production deployment requires a few steps.
+
+Step 1: Save the following to your-values.yaml, ideally your git repo.
```
Comment (faisal-memon) [inline charts/spire/README.md:24]: can we use the markdown numbers?

Comment (faisal-memon) [inline charts/spire/README.md:24]:
```suggestion
Step 1: Save the following to your-values.yaml, ideally in your git repo.
```

File: charts/spire/README.md [hunk 7]
```diff
@@ -7,33 +7,76 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
...
+    # Update these
+    country: ARPA
+    organization: Example
+    common_name: example.org
+```
+
+Step 2: If you need a non default storageClass, append the following to the spire-server section and update:
+```
+  persistence:
+    storageClass: changeme
```
Comment (faisal-memon) [inline charts/spire/README.md:48]:
```suggestion
    storageClass: your-storage-class
```

File: charts/spire/README.md [hunk 8]
```diff
@@ -7,33 +7,76 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
...
 
+Step 3: If your Kubernetes cluster is OpenShift based, use the output of the following command for your trustDomain:
 ```shell
-curl -o your-values.yaml https://raw.githubusercontent.com/spiffe/helm-charts-hardened/main/examples/production/example-your-values.yaml
+oc get cm -n openshift-config-managed  console-public -o go-template="{{ .data.consoleURL }}" | sed 's@https://@@; s/^[^.]*\.//'
 ```
 
-Then:
+Step 4: Find any additional values you might want to set based on the documentation below or the examples at:
+https://github.com/spiffe/helm-charts-hardened/tree/main/examples
```
Comment (faisal-memon) [inline charts/spire/README.md:57]:
```suggestion
Step 4: Find any additional values you might want to set based on the documentation below or using the [examples](https://github.com/spiffe/helm-charts-hardened/tree/main/examples)
```

File: charts/spire/README.md [hunk 9]
```diff
@@ -7,33 +7,76 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
...
+
+In particular, consider using an external database.
+
+Step 5: Edit your-values.yaml with the appropriate values.
```
Comment (faisal-memon) [inline charts/spire/README.md:61]: This step doesnt seem to have anything under it.

File: charts/spire/README.md [hunk 10]
```diff
@@ -7,33 +7,76 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
...
+
+Step 6: Deployment
```
Comment (faisal-memon) [inline charts/spire/README.md:63]:
```suggestion
Step 6: Deploy
```

File: charts/spire/README.md [hunk 11]
```diff
@@ -7,33 +7,76 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
...
 
-For production installs, please see [the production example](https://github.com/spiffe/helm-charts-hardened/tree/spire-0.16.0/examples/production).
-
 ## Upgrade notes
 
 We only support upgrading one major version at a time. Version skipping isn't supported.
 
+### 0.17.X
+
+The SPIFFE OIDC Discovery Provider now has many new TLS options and out of the box defaults to using SPIRE for its certificate. Also, the `spiffe-oidc-discovery-provider.insecureScheme.enabled` flag was removed. If you previously set that option, remove the setting from your values.yaml and see if the new default is suitable for your deployment. If it isn't, please consider one of the other options under `spiffe-oidc-discovery-provider.tls`. If all other options are still unsuitable, you can still enable the previous mode by disabling all tls options. (`spiffe-oidc-discovery-provider.spire.enabled=false`)
```
Comment (faisal-memon) [inline charts/spire/README.md:76]:
```suggestion
- The SPIFFE OIDC Discovery Provider now has many new TLS options and defaults to using SPIRE to issue its certificate.
- The `spiffe-oidc-discovery-provider.insecureScheme.enabled` flag was removed. If you previously set that flag, remove the setting from your values.yaml and see if the new default of using a SPIRE issued certificate is suitable for your deployment. If it isn't, please consider one of the other options under `spiffe-oidc-discovery-provider.tls`. If all other options are still unsuitable, you can still enable the previous mode by disabling TLS. (`spiffe-oidc-discovery-provider.spire.enabled=false`)
```

File: charts/spire/README.md [hunk 12]
```diff
@@ -7,33 +7,76 @@ A Helm chart for deploying the complete Spire stack including: spire-server, spi
...
+
+The SPIFFE OIDC Discovery Provider is now enabled by default. If you previously chose to have it off, you can disable it explicitly with `spiffe-oidc-discovery-provider.enabled=false`.
```
Comment (faisal-memon) [inline charts/spire/README.md:78]:
```suggestion
- The SPIFFE OIDC Discovery Provider is now enabled by default. If you previously chose to have it off, you can disable it explicitly with `spiffe-oidc-discovery-provider.enabled=false`.
```

File: charts/spire/README.md (hunk 6 above)
Comment (kfox1111) [reply charts/spire/README.md:24]: I think markdown numbers only work when they are in a direct list, like? :
```
1. thing a
1. thing b
1. thing c
```

File: charts/spire/README.md (hunk 9 above)
Comment (kfox1111) [reply charts/spire/README.md:61]: The other steps are about gathering the options you might need, and this is talking about actually making the changes.

So, something else above is unclear I guess.

File: charts/spire/README.md (hunk 6 above)
Comment (faisal-memon) [reply charts/spire/README.md]: Yes, is it possible to switch to that numbering?

**Review by faisal-memon — APPROVED** [review]

**Pull request merged**